crates/rspack_binding_options/src/js_values/union.rs

```diff
@@ -0,0 +1,59 @@
+use std::fmt::Debug;
+
+use napi::{NapiRaw, NapiValue};
+
+/// The concept is similar to `Either`, but `Either` is a common used name. To avoid conflictcations I
+/// borrow the `Union` name from the `union type` of TypeScript.
+pub enum Union<A, B> {
+  Left(A),
+  Right(B),
+}
+
+impl<Left, Right> Union<Left, Right> {
+  pub fn expect_left(self) -> Left {
+    match self {
+      Union::Left(l) => l,
+      Union::Right(_) => panic!("expect left, but got right"),
+    }
+  }
+  pub fn expect_right(self) -> Right {
+    match self {
+      Union::Left(_) => panic!("expect right, but got left"),
+      Union::Right(r) => r,
+    }
+  }
+}
+
+// TODO(hyf0): this is really not ideal. I will fix it when I understand
+// https://users.rust-lang.org/t/check-if-a-trait-is-implemented-or-not/73756
+impl<A, B> Debug for Union<A, B> {
```

Comment:
Why not `impl<A: Debug, B: Debug> Debug for Union<A, B> {`

Reply:
`JsString` doesn't implement `Debug`.

Reply:
This does not have anything to do with the concrete type actually.